### ISSUE TITLE
Implement sentiment reward

### ISSUE THREAD

**Issue by diogo-cruz** [EdoardoPona/predicting-inductive-biases-RL#12]
This consists of:

1. Loading some finetuned ML model for sentiment analysis.
2. Taking prompt+generated_sequence as input for that model and outputting a scalar value that can act as a reward for PPO.

**Comment by EdoardoPona** [comment]
### Pretrained model for sentiment analysis 

We could use https://huggingface.co/cardiffnlp/twitter-roberta-base-sentiment-latest
tldr: trained on tweets for 3 classes:  negative, neutral, positive.

reward can be the cross entropy with the desired class.

**Comment by EdoardoPona** [comment]
current implementation at https://github.com/allenai/RL4LMs/commit/e147dd3dee0e539cb96d97e3ffb851ef63c85fc0

largely untested end to end, but individually works

**Comment by EdoardoPona** [comment]
current reward model was trained on twitter data. Given meeting outcome, agreeing to fine-tune on same dataset as sentiment model, I will update this to use a model based on IMDB.

**Comment by EdoardoPona** [comment]
implemented in https://github.com/allenai/RL4LMs/commit/d80687c8bf51e062665f7c9dc5aa5dfd76aebf46#diff-536f6459cbccfcd83858e1758cf46ab592a5393e506cc124e9c32116673bfb28 

features
- generic abstract learned classifier reward which holds any pretrained classifier loadable by huggingface automodels. 
- concrete bert twitter and xlnet imdb reward function implementations 
- metrics for implemented rewards
